HPXMLtoOpenStudio/resources/electric_panel.rb

```diff
@@ -191,17 +186,30 @@ def self.calculate_meter_based(hpxml_bldg, electric_panel, service_feeders, serv
   # @param electric_panel [HPXML::ElectricPanel] Object that defines a single electric panel
   # @param [Array<HPXML::ServiceFeeder>] List of service feeder objects
   # @return [nil]
-  def self.calculate_breaker_spaces(electric_panel, service_feeders)
-    occupied = electric_panel.branch_circuits.map { |branch_circuit| branch_circuit.occupied_spaces }.sum(0.0)
+  def self.calculate_breaker_spaces(electric_panel)
+    occupied = get_occupied_spaces(electric_panel)
+
     if not electric_panel.rated_total_spaces.nil?
       total = electric_panel.rated_total_spaces
     else
-      total = occupied + electric_panel.headroom_spaces
+      total = occupied + electric_panel.headroom_spaces # headroom is either specified or 3
+
+      electric_panel.rated_total_spaces = total
+      electric_panel.rated_total_spaces_isdefaulted = true
+    end
+
+    if electric_panel.headroom_spaces.nil? # only nil if total is specified
+      electric_panel.headroom_spaces = total - occupied
+      electric_panel.headroom_spaces_isdefaulted = true
     end
+  end
 
-    service_feeders.BreakerSpaces_Total = total
-    service_feeders.BreakerSpaces_Occupied = occupied
-    service_feeders.BreakerSpaces_HeadRoom = total - occupied
+  # Calculate the sum of OccupiedSpaces across BranchCircuits.
+  #
+  # @param electric_panel [HPXML::ElectricPanel] Object that defines a single electric panel
+  # @return [Double] Total occupied spaces
+  def self.get_occupied_spaces(electric_panel)
+    return electric_panel.branch_circuits.map { |branch_circuit| branch_circuit.occupied_spaces }.sum(0.0)
```

Comment:
Wrapping this in a method because it gets used in several places.

Reply:
https://github.com/NREL/OpenStudio-HPXML/pull/2091/commits/f5d6b8311fdc2ba3ba61aa47a716e49f9237c857